c7n/reports/csvout.py

```diff
@@ -263,44 +266,56 @@ def record_set(session_factory, bucket, key_prefix, start_date):
     """
 
     s3 = local_session(session_factory).client('s3')
-    marker = key_prefix.strip("/") + "/" + start_date.strftime(
-        '%Y-%m-%d-00') + "/resources.json.gz"
+
+    today = date.today()
+    current_date = start_date.date()
+    date_folders = []
```

Comment:
this feels like your doing too much work, s3 buckets are listed alphabetically.. given a start date folder, you get lexographical sorting to today.